database/db.go

```diff
@@ -17,28 +19,50 @@ type dbConfig struct {
 	password string
 }
 
-var config = dbConfig{
-	os.Getenv("POSTGRES_HOST"),
-	os.Getenv("POSTGRES_PORT"),
-	os.Getenv("POSTGRES_USER"),
-	os.Getenv("POSTGRES_DB"),
-	os.Getenv("POSTGRES_PASSWORD"),
+func getConfig() dbConfig {
+	return dbConfig{
+		os.Getenv("POSTGRES_HOST"),
+		os.Getenv("POSTGRES_PORT"),
+		os.Getenv("POSTGRES_USER"),
+		os.Getenv("POSTGRES_DB"),
+		os.Getenv("POSTGRES_PASSWORD"),
+	}
 }
 
-var dns = fmt.Sprintf(
-	"host=%s port=%s user=%s dbname=%s password=%s sslmode=disable",
-	config.host, config.port, config.user, config.dbname, config.password)
+func getTestConfig() dbConfig {
+	return dbConfig{
+		os.Getenv("POSTGRES_HOST"),
+		os.Getenv("POSTGRES_TEST_PORT"),
+		os.Getenv("POSTGRES_USER"),
+		os.Getenv("POSTGRES_TEST_DB"),
+		os.Getenv("POSTGRES_PASSWORD"),
+	}
+}
+
+func getDns(config dbConfig) string {
+	return fmt.Sprintf(
+		"host=%s port=%s user=%s dbname=%s password=%s sslmode=disable",
+		config.host, config.port, config.user, config.dbname, config.password)
+}
 
 func Init() error {
-	DB, err := gorm.Open(postgres.Open(dns), &gorm.Config{})
+	config := getConfig()
+	dns := getDns(config)
+	db, err := gorm.Open(postgres.Open(dns), &gorm.Config{})
 	if err != nil {
 		return err
 	}
+	DB = db
+	return nil
+}
 
-	err = initGormUserStore(DB)
+func InitTest() error {
+	config := getTestConfig()
+	dns := getDns(config)
+	db, err := gorm.Open(postgres.Open(dns), &gorm.Config{})
 	if err != nil {
 		return err
 	}
-
+	DB = db
 	return nil
 }
```

Comment:
Same question: why are tests baked in?

Reply:
[Ditto here](https://github.com/leaderboardsgg/leaderboard-backend/pull/78/files#r777206030). (Just to repeat: this isn't for testing the `db` package itself, but for other packages, namely `user`)